google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -707,6 +708,88 @@ public void testCreateTableWithRangePartitioning() {
+          Collections.singletonMap(
+              "jsonField", "{\"student\" : {\"name\" : \"Alice\", \"id\": 12}}");
+      Map<String, Object> jsonRow4 =
+          Collections.singletonMap(
+              "jsonField", "{\"student\" : {\"name\" : \"Bijoy\", \"id\": 14}}");
+      InsertAllRequest request =
+          InsertAllRequest.newBuilder(tableId)
+              .addRow(jsonRow1)
+              .addRow(jsonRow2)
+              .addRow(jsonRow3)
+              .addRow(jsonRow4)
+              .build();
+      InsertAllResponse response = bigquery.insertAll(request);
+      assertFalse(response.hasErrors());
+      assertEquals(0, response.getInsertErrors().size());
+
+      // Query the JSON column with string positional query parameter
+      String sql =
+          "SELECT jsonField.class.student.id FROM "
+              + tableId.getTable()
+              + " WHERE JSON_VALUE(jsonField, \"$.class.student.name\")  = ? ";
+      QueryParameterValue stringParameter = QueryParameterValue.string("Jane");
+      QueryJobConfiguration queryJobConfiguration =
+          QueryJobConfiguration.newBuilder(sql)
+              .setDefaultDataset(DatasetId.of(DATASET))
+              .setUseLegacySql(false)
+              .addPositionalParameter(stringParameter)
+              .build();
+      TableResult result = bigquery.query(queryJobConfiguration);
+      for (FieldValueList values : result.iterateAll()) {
+        assertEquals("10", values.get(0).getValue());
+      }
+
+      // Insert another JSON row parsed from a String with json positional query parameter
+      String dml = "INSERT INTO " + tableId.getTable() + " (jsonField) VALUES(?)";
+      QueryParameterValue jsonParameter =
+          QueryParameterValue.json("{\"class\" : {\"student\" : [{\"name\" : \"Amy\"}]}}");
+      QueryJobConfiguration dmlQueryJobConfiguration =
+          QueryJobConfiguration.newBuilder(dml)
+              .setDefaultDataset(DatasetId.of(DATASET))
+              .setUseLegacySql(false)
+              .addPositionalParameter(jsonParameter)
+              .build();
+      bigquery.query(dmlQueryJobConfiguration);
+      Page<FieldValueList> rows = bigquery.listTableData(tableId);
+      assertEquals(5, Iterables.size(rows.getValues()));
+
+      // Insert another JSON row parsed from a JsonObject with json positional query parameter
+      JSONObject jsonObject = new JSONObject().put("class", "student");
+      QueryParameterValue jsonParameter1 = QueryParameterValue.json(jsonObject);
+      QueryJobConfiguration dmlQueryJobConfiguration1 =
+          QueryJobConfiguration.newBuilder(dml)
+              .setDefaultDataset(DatasetId.of(DATASET))
+              .setUseLegacySql(false)
+              .addPositionalParameter(jsonParameter1)
+              .build();
+      bigquery.query(dmlQueryJobConfiguration1);
+      Page<FieldValueList> rows1 = bigquery.listTableData(tableId);
+      assertEquals(6, Iterables.size(rows1.getValues()));
```

Comment:
Might be nice to verify that the value we get back for the `jsonField` column has the expected value.

Though hopefully the BigQuery backend at least verifies that we have inserted correct JSON syntax?

That might be an interesting test: verify that it fails if we try to send a string value that is not valid JSON.

Reply:
Added assert to verify JSON content.

Added another test to verify that malformed JSON is rejected.